plugin/evm/limit_order.go

```diff
@@ -128,30 +129,45 @@ func (lop *limitOrderProcesser) runMatchingEngine(longOrders []limitorders.Limit
 }
 
 func (lop *limitOrderProcesser) runLiquidations(market limitorders.Market, longOrders []limitorders.LimitOrder, shortOrders []limitorders.LimitOrder) (filteredLongOrder []limitorders.LimitOrder, filteredShortOrder []limitorders.LimitOrder) {
```

Comment:
while this works, but is a clone of the matching logic. How about this approach:
- place in-memory market orders on behalf of the account being liquidated.
- Will need a "Liquidation: true" flag or smth
- Let it get matched through normal matching logic
- If 1 of the matched orders has `liquidation: true` set, call `ExecuteLiquidation` otherwise `ExecuteMatchedOrder`
- Purge the in-memory liquidation orders

Reply:
I tried doing it this way too, but changed it back because it was looking too complicated to have liquidations run through matching engine and to have positions in LimitOrder struct. Also, it didn't solve the problem of doing all long positions before short positions. That's why I've kept liquidations function separate